common/client-sdk/rest_client.go

```diff
@@ -0,0 +1,357 @@
+package arkclient
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	arkv1 "github.com/ark-network/ark/api-spec/protobuf/gen/ark/v1"
+)
+
+func NewRestClient(baseURL string, httpClient *http.Client) (ArkRestClient, error) {
+	if baseURL == "" {
+		return nil, errAspUrlEmpty
+	}
+
+	return &arkRestClient{baseURL: baseURL, httpClient: httpClient}, nil
+}
+
+type ArkRestClient interface {
+	Admin() AdminServiceClient
+	Service() ArkServiceClient
+}
+
+type AdminServiceClient interface {
+	GetBalance(
+		ctx context.Context, req *arkv1.GetBalanceRequest,
+	) (*arkv1.GetBalanceResponse, error)
+	GetScheduledSweep(
+		ctx context.Context, req *arkv1.GetScheduledSweepRequest,
+	) (*arkv1.GetScheduledSweepResponse, error)
+	GetRoundDetails(
+		ctx context.Context, req *arkv1.GetRoundDetailsRequest,
+	) (*arkv1.GetRoundDetailsResponse, error)
+	GetRounds(
+		ctx context.Context, req *arkv1.GetRoundsRequest,
+	) (*arkv1.GetRoundsResponse, error)
+}
+
+type ArkServiceClient interface {
+	RegisterPayment(
+		ctx context.Context, req *arkv1.RegisterPaymentRequest,
+	) (*arkv1.RegisterPaymentResponse, error)
+	ClaimPayment(
+		ctx context.Context, req *arkv1.ClaimPaymentRequest,
+	) (*arkv1.ClaimPaymentResponse, error)
+	FinalizePayment(
+		ctx context.Context, req *arkv1.FinalizePaymentRequest,
+	) (*arkv1.FinalizePaymentResponse, error)
+	GetRound(
+		ctx context.Context, req *arkv1.GetRoundRequest,
+	) (*arkv1.GetRoundResponse, error)
+	GetEventStream(
+		ctx context.Context, req *arkv1.GetEventStreamRequest,
+	) (arkv1.ArkService_GetEventStreamClient, error)
+	Ping(
+		ctx context.Context, req *arkv1.PingRequest,
+	) (*arkv1.PingResponse, error)
+	ListVtxos(
+		ctx context.Context, req *arkv1.ListVtxosRequest,
+	) (*arkv1.ListVtxosResponse, error)
+	GetInfo(
+		ctx context.Context, req *arkv1.GetInfoRequest,
+	) (*arkv1.GetInfoResponse, error)
+	Onboard(
+		ctx context.Context, req *arkv1.OnboardRequest,
+	) (*arkv1.OnboardResponse, error)
+	TrustedOnboarding(
+		ctx context.Context, req *arkv1.TrustedOnboardingRequest,
+	) (*arkv1.TrustedOnboardingResponse, error)
+}
+
+type arkRestClient struct {
+	baseURL    string
+	httpClient *http.Client
+}
+
+func (c *arkRestClient) Admin() AdminServiceClient {
+	return &restAdminServiceClient{baseURL: c.baseURL, client: c.httpClient}
+}
+
+func (c *arkRestClient) Service() ArkServiceClient {
+	return &restArkServiceClient{baseURL: c.baseURL, client: c.httpClient}
+}
+
+type restAdminServiceClient struct {
```

Comment:
is this all written by hand? NO way to do code-gen from Swagger?

Reply:
https://goswagger.io/go-swagger/about/

Reply:
I will check this, code was written by hand.